### PR TITLE
Working substring for Author and Keyword

### PR DESCRIPTION
**Pull request by limmabean**
This does not work for Tags, but since tags are single phrases anyway, it shouldn't be hard to search the full tag.

### CODE REVIEW

**Comment by karpawich** [comment]
[Search bar does not return substring matches, just word matches.](https://trello.com/c/Xe9EJet3/78-search-bar-does-not-return-substring-matches-just-word-matches)

File: models/query/report-query.js [hunk 1]
```diff
@@ -68,15 +69,19 @@ ReportQuery.prototype.toMongooseFilter = function() {
   // Determine author filter
   if (this.author) {
     filter.author = {};
-    filter.author.$in = toRegexp.alli(this.author.trim().split(/\s*,\s*/).sort());
+    filter.author.$options = 'i';
+    filter.author.$regex = this.author;
```
Comment (cooperka) [inline models/query/report-query.js:72]: Would love a basic explanation of what this did before and what it does now. I'm curious why they were all turned into an array before if it's as simple as using $regex instead. Does this mean we can only search one author at a time now?

File: models/query/report-query.js [hunk 2]
```diff
@@ -68,15 +69,19 @@ ReportQuery.prototype.toMongooseFilter = function() {
...
   }
   if (this.tags) {
-    filter.tags = { $all: toRegexp.allCaseInsensitive(this.tags) };
```
Comment (cooperka) [inline models/query/report-query.js:74]: Looks like the only remaining `toRegexp` usage after this is in `incident.js`, can we get rid of that too and delete the whole `toRegexp` file? Would be a nice cleanup, that file seems smelly to me.

**Review by cooperka — APPROVED** [review]

**Review by TheDanieLNK — APPROVED** [review]
Looks good to me. Thanks Matt.

File: models/query/report-query.js (hunk 1 above)
Comment (limmabean) [reply models/query/report-query.js:72]: Previously we used a handwritten method of regexing the inputs. However, mongoDB now supports built in regex functions, so this is unecessary. Somewhere along the way, this previous method stopped working very well.

Comment (limmabean) [reply models/query/report-query.js:72]: Also yes, we can only search one author at a time.

**Comment by limmabean** [comment]
Further Improvements include standardizing the search architecture between incidents and reports and supporting multiple searches in bars. This is currently limited because we don't use $or or $and functionality in our filters. This is not difficult to implement if we have a predetermined set of delimiters and indicators of $and and $or searches (eg. "Car + Tire" searches for posts with "Car" and "Tire")

**Pull request merged**